polars/polars-io/src/csv/write_impl.rs

```diff
@@ -329,17 +329,24 @@ pub(crate) fn write<W: Write>(
 
         n_rows_finished += total_rows_per_pool_iter;
     }
-
     Ok(())
 }
+
 /// Writes a CSV header to `writer`
 pub(crate) fn write_header<W: Write>(
     writer: &mut W,
     names: &[&str],
     options: &SerializeOptions,
 ) -> PolarsResult<()> {
+    let mut escaped_names: Vec<String> = Vec::with_capacity(names.len());
+    let mut nm: Vec<u8> = Vec::with_capacity(1);
+    for name in names {
+        fmt_and_escape_str(&mut nm, name, options)?;
+        escaped_names.push(std::str::from_utf8(&nm).unwrap().to_string());
+        nm.clear();
+    }
     writer.write_all(
-        names
+        escaped_names
             .join(std::str::from_utf8(&[options.delimiter]).unwrap())
```

Comment:
we could use `from_utf8_unchecked` as we know we are dealing with utf8 data here. This might be worth it for the genomics people who deal with millions of columns.

Reply:
Done 👍